h/api_client/api_client.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+import requests
+
+
+class APIError(Exception):
+
+    """Raised when a request to the API fails for any reason."""
+
+    pass
+
+
+class ConnectionError(APIError):
+
+    """Raised when a request to the API fails because of a network problem."""
+
+    pass
+
+
+class Timeout(APIError):
+
+    """Raised when a request to the API times out."""
+
+    pass
+
+
+class Client(object):
+
+    """A client for the Hypothesis API.
+
+    Provides simple methods for calling the API over HTTP.
+
+    Note that this class is deliberately decoupled from the Hypothesis Pyramid
+    app and from Pyramid.
+
+    """
+
+    def __init__(self, base_url):
+        """Initialize a new API client.
+
+        :param base_url: The base URL to the Hypothesis API instance to use.
+            For example: ``"http://hypothesis.is/api"``.
+
+        """
+        self.base_url = base_url
+
+    def get(self, path, params=None):
```

Comment:
I think we evolve it as we use it. There's no point disappearing off into abstraction land when we only use this in a single place in our code :).